### DISCUSSION TITLE
LED Controller setup with Adafruit ItsyBitsy RP2040

### DISCUSSION THREAD

**Discussion by guyisme** [awawa-dev/HyperSerialPico#24] [Q&A]
Hello. I'm having difficulty to setup led controller. I have Adafruit ItsyBitsy RP2040 and SK9822 rgb leds. I could succesfuly flash the firmware in RP2040 but I cannot see any USB device in output path as shown in picture. I also tried to flash with optional workaround for non booting boards but nothing has changed. What could be wrong?

Thanks in advance.

Umit

![ItsyBitsy](https://github.com/awawa-dev/HyperSerialPico/assets/114879039/6f0d3334-2f47-47a2-853a-edcd9ef94e27)

**Comment by guyisme** [comment]
Sorry, correct picture is here...

![ItsyBitsy2](https://github.com/awawa-dev/HyperSerialPico/assets/114879039/31c13ec9-8929-484b-904f-6b604cb3c408)

**Comment by awawa-dev** [comment]
Hi
Upload full logs at pastebin.com service or something similar taken after 2-3 minutes after HyperHDR start. What is your OS/device? If you disconnect rp2040 from USB port does the list of 'output path' changes?

**Comment by guyisme** [comment]
I'm running raspberry pi 4 and installed SD-card-image-32bits-armv6l.img. You can find the log in https://justpaste.it/cb4po .

When I disconnect rp2040 usb cable, I got the following in log:

2024-04-27T16:19:47.657Z [LEDDEVICE0_ADALIGHT] (ProviderSerial.cpp:164) Serial port is closed: ttyACM0
2024-04-27T16:19:47.657Z [LEDDEVICE0_ADALIGHT] (LedDevice.cpp:381) Stopping refresh timer
2024-04-27T16:19:47.657Z [LEDDEVICE0_ADALIGHT] Device 'adalight' is disabled due to an error: 'Rs232 SerialPortError: Resource temporarily unavailable'
2024-04-27T16:19:47.659Z [COMPONENTCTRL0] LED device: disabled
2024-04-27T16:19:47.659Z [PERFORMANCE] [USB: FPS = 30.01, decoding = 8ms, frames = 1800, invalid = 0, mode = direct], [INSTANCE0: FPS = 30.01, processed = 1800]

**Comment by guyisme** [comment]
Output patch screenshot after unplugging usb cable:

![ItsyBitsy3](https://github.com/awawa-dev/HyperSerialPico/assets/114879039/26679b6e-286e-4be3-91b3-a1b101c95f42)

**Comment by awawa-dev** [comment]
Seems you rp2040 is detected correctly:
`2024-04-27T15:51:46.012Z [LEDDEVICE0_ADALIGHT] Received goodbye: 'HyperHDR frames: 0 (FPS), receiv.: 0, good: 0, incompl.: 0, mem1: 0, mem2: 460, heap: 123232`
`2024-04-27T15:51:46.514Z [LEDDEVICE0_ADALIGHT] DETECTED DEVICE USING HyperSerialEsp8266/HyperSerialESP32/HyperSerialPico FIRMWARE`
so it should work although 0x1106 0x3483 are not correct productId/vendorId (must use force HyperSerial detection as a workaround) and it indicates UDev bug so try to update your system first (apt update & apt upgrade)

**Reply by guyisme** [reply]
Thank you for your suggestion, I applied system update & upgrade both. Still cannot see usb device in the list!. You can see the new log after rebooting the system here: https://justpaste.it/9jsat

**Reply by awawa-dev** [reply]
I forgot that for RPi4 I added workaround is in the HyperHDR so you dont need use 'Force HyperSerial detection'. This is PR: https://github.com/awawa-dev/HyperHDR/pull/583
and code:
`
else if (serialPortInfo.productIdentifier() == 0x3483 && serialPortInfo.vendorIdentifier() == 0x1106)
`
But I think this UDev bug is already fixed at least in aarch64 OS (I dont use legacy arm6 architecture). You may not see 'Pico' label when UDev bug is present but your device is initialized correctly:
```
2024-04-27T16:41:34.615Z [LEDDEVICE0_ADALIGHT] Enabling the Rpi4 udev bug workaround. The serial device is incorrectly identified by the OS and HyperHDR skips the reset. State: 1, 1
2024-04-27T16:41:34.617Z [LEDDEVICE0_ADALIGHT] ESP sent: 'HyperHDR frames: 0 (FPS), receiv.: 0, good: 0, incompl.: 0, mem1: 0, mem2: 460, heap: 123232'
2024-04-27T16:41:34.617Z [LEDDEVICE0_ADALIGHT] DETECTED DEVICE USING HyperSerialEsp8266/HyperSerialESP32/HyperSerialPico FIRMWARE (Welcome! Awa driver 9.) at 1 msec
```

**Reply by guyisme** [reply]
Ok, happy to hear the device is initialized, thank you very much. I will also install aarch64 version (previously I had no luck but probably due to sd card issue).

I have one more question. I have both Ezcap 269 and MS2130 (newer firmware). My media source is Nvidia Shield TV. Which capture card you would suggest to use in my setup?

**Reply by awawa-dev** [reply]
Definitely ms2130, the image quality in ezcap 269 is worse. Its only advantage used to be that it allowed HDR passthrough as a 2in1 solution (but only in the selected color space, which it also switched when it started capturing), but now there are a lot of cheap HDMI splitters and much better grabbers like ms2130. For ms2130 there are 2 LUTs, Shield uses limited colorspace, so you should choose a LUT with limited colorspace, otherwise you may have problems with obtaining a black color. Anyway you can try both.

**Reply by guyisme** [reply]
Thank you very much for your help, I will try both tables. Have a nice weekend...
